src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -488,14 +493,59 @@ extern "C" bool PalDetachThread(void* thread)
 }
 
 #if !defined(USE_PORTABLE_HELPERS) && !defined(FEATURE_RX_THUNKS)
+
+#ifdef TARGET_APPLE
+static const struct section_64 *thunks_section;
+static const struct section_64 *thunks_data_section;
+#endif
+
 REDHAWK_PALEXPORT UInt32_BOOL REDHAWK_PALAPI PalAllocateThunksFromTemplate(HANDLE hTemplateModule, uint32_t templateRva, size_t templateSize, void** newThunksOut)
 {
+#ifdef TARGET_APPLE
+    int f;
+    char *exepath;
+
+    // NOTE: We ignore hTemplateModule, it is always the current module
+    exepath = minipal_getexepath();
```

Comment:
This won't work for dynamic libraries. Is there an API to get the path to the executable or .dylib that this code lives in?

Reply:
I will look into it but I am not aware of any easy way to accomplish that. The `getsect` APIs have a variant with pointer to Mach-O header, and technically the unwinding code should be able to resolve it somehow based on the current frame.

Reply:
`dladdr` + `dli_fname`? Native AOT PAL has existing uses of it.

Reply:
That looks promising. Thanks for the idea.

Reply:
Updated the code. I only tested it on the previous scenarios with an executable but it didn't break there.